File: 文件夹/C#/排序/选择排序.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class 选择排序 : MonoBehaviour
{
    void SelectSort(int[] dataArray)
    {
        for (int i = 0; i < dataArray.Length - 1; i++)
        {
            int min = dataArray[i];
            int minIndex = i;
            for (int j = i + 1; j < dataArray.Length; j++)
            {
                if (dataArray[j] < min)
                {
                    min = dataArray[j];
                    minIndex = j;
                }
            }
            if (minIndex != i)
            {
                int temp = dataArray[i];
                dataArray[i] = dataArray[minIndex];
                dataArray[minIndex] = temp;
            }
        }
    }
}

File: 文件夹/Lua/排序/选择排序.lua
function ChoiceSort(array)
	for i=1,#array -1 do
		local min = array[i]
		local minIndex = i
		for j=i+1,#array do
			if(array[j]<min) then
				minIndex = j
			end
		end
		if(minIndex ~= min) then
			local temp = array[minIndex]
			array[minIndex] = array[i]
			array[i] = temp
		end
	end
	return array
end
--array1ÅÅÐò
local array1 = {73,23,18,92}
ChoiceSort(array1, 1,#array1)
--array2ÅÅÐò
array2 = {73,23,18,92}
table.sort(array2)
--array1ÅÅÐòÊä³ö
print("array1ÅÅÐòÊä³ö")
for k,v in pairs(array1) do
	print(v)
end
--array2ÅÅÐòÊä³ö
print("array2ÅÅÐòÊä³ö")
for k,v in pairs(array2) do
	print(v)
end

